symphony/app/fbcnms-projects/workflows/src/routes.js

```diff
@@ -20,553 +20,570 @@ import moment from 'moment';
+      }
+    },
+  );
 
-const findSchedule = (schedules, name) => {
-  for (const schedule of schedules) {
-    if (schedule.name === name) {
-      return schedule;
+  router.put('/metadata', async (req: ExpressRequest, res, next) => {
+    try {
+      const result = await http.put(baseURLMeta + 'workflow/', req.body, req);
+      res.status(200).send(result);
+    } catch (err) {
+      res.status(400).send(err.response.body);
+      next(err);
     }
-  }
-  return null;
-};
+  });
+
+  // Conductor only allows POST for event handler creation
+  // and PUT for updating. This code works around the issue by
+  // trying PUT if POST fails.
+  router.post('/event', async (req: ExpressRequest, res, next) => {
+    try {
+      const result = await http.post(eventURL, req.body, req);
+      res.status(200).send(result);
+    } catch (err) {
+      logger.info(`Got exception ${JSON.stringify(err)} while POSTing event`);
+      try {
+        const result = await http.put(eventURL, req.body, req);
+        res.status(200).send(result);
+      } catch (e) {
+        logger.info(`Got exception ${JSON.stringify(e)} while PUTting event`);
+        res.status(400).send('Post and Put failed');
+        next(e);
+      }
+    }
+  });
 
-router.get('/metadata/workflow', async (req: ExpressRequest, res, next) => {
-  try {
-    const result = await http.get(baseURLMeta + 'workflow', req);
-    // combine with schedules
-    const schedules = await http.get(baseURLSchedule, req);
-    for (const workflowDef of result) {
-      const expectedScheduleName = workflowDef.name + ':' + workflowDef.version;
-      const found = findSchedule(schedules, expectedScheduleName);
-      workflowDef.hasSchedule = found != null;
-      workflowDef.expectedScheduleName = expectedScheduleName;
+  router.post('/workflow', async (req: ExpressRequest, res, next) => {
+    try {
+      const result = await http.post(baseURLWorkflow, req.body, req);
+      res.status(200).send(result);
+    } catch (err) {
+      next(err);
     }
+  });
 
-    res.status(200).send({result});
-  } catch (err) {
-    next(err);
-  }
-});
+  router.get('/executions', async (req: ExpressRequest, res, next) => {
+    try {
+      const freeText = [];
+      if (req.query.freeText !== '') {
+        freeText.push(req.query.freeText);
+      } else {
+        freeText.push('*');
+      }
 
-router.delete(
-  '/metadata/workflow/:name/:version',
-  async (req: ExpressRequest, res, next) => {
+      let h: string = '-1';
+      if (req.query.h !== 'undefined' && req.query.h !== '') {
+        /* $FlowFixMe req.query is user-controlled input, properties and values
+         in this object are untrusted and should be validated before trusting */
+        h = req.query.h;
+      }
+      if (h !== '-1') {
+        freeText.push('startTime:[now-' + h + 'h TO now]');
+      }
+      let start: number = 0;
+      if (!isNaN(req.query.start)) {
+        // $FlowFixMe: isNaN is sketchy and accepts arrays
```

Comment:
Is "FlowFixMe" required here for flow? This doesn't look like a flow issue (but I agree that there should be a fixme comment)

Reply:
I noticed this isn't your change, so feel free to leave it as-is. Maybe make a note to change it later.

Reply:
Correction - the linter is complaining about the `$FlowFixMe` comments, eslint has a "no-warning-comments" rule. There's 6 occurences. I'm fine if you change it to `// FixMe`.

Reply:
There's was a change 6 days ago to run the linter in CircleCI, not sure why it isn't showing up here, I'll look into that.

Reply:
Changed